unleash/Unleash.swift

```diff
@@ -9,93 +9,123 @@ import Foundation
-    public private(set) var url: String
-    public private(set) var refreshInterval: Int?
-    public private(set) var strategies: [Strategy]
+    self.init(
+      clientRegistration: clientRegistration,
+      registerService: registerService,
+      toggleRepository: toggleRepository,
+      appName: appName,
+      url: url,
+      refreshInterval: refreshInterval,
+      strategies: allStrategies)
+  }
+  
+  init(
+    clientRegistration: ClientRegistration,
+    registerService: RegisterServiceProtocol,
+    toggleRepository: ToggleRepositoryProtocol,
+    appName: String,
+    url: String,
+    refreshInterval: Int?,
+    strategies: [Strategy]
+  ) {
+    self.registerService = registerService
+    self.toggleRepository = toggleRepository
+    self.appName = appName
+    self.url = url
+    self.refreshInterval = refreshInterval
+    self.strategies = strategies
     
-    public convenience init(appName: String, url: String, refreshInterval: Int?, strategies: [Strategy] = []) {
-        let clientRegistration: ClientRegistration = ClientRegistration(appName: appName, strategies: strategies)
-        let registerService: RegisterServiceProtocol = RegisterService()
-        let memory: MemoryCache = MemoryCache(cache: Cache(), jsonDecoder: JSONDecoder(), jsonEncoder: JSONEncoder())
-        let toggleService: ToggleServiceProtocol
-            = ToggleService(appName: appName, instanceId: clientRegistration.instanceId)
-        let toggleRepository: ToggleRepository = ToggleRepository(memory: memory, toggleService: toggleService)
-        let allStrategies: [Strategy] = [DefaultStrategy()] + strategies
-
-        self.init(clientRegistration: clientRegistration, registerService: registerService,
-                  toggleRepository: toggleRepository, appName: appName, url: url, refreshInterval: refreshInterval,
-                  strategies: allStrategies)
+    register(body: clientRegistration)
+    .then { self.toggleRepository.get(url: URL(string: url)!) }
+    .done { _ in self.delegate?.unleashDidFetchToggles(self) }
+    .catch { error in
+      log("error \(error)")
+      self.delegate?.unleash(self, didFailWithError: error)
     }
-    
-    init(clientRegistration: ClientRegistration, registerService: RegisterServiceProtocol,
-         toggleRepository: ToggleRepositoryProtocol, appName: String, url: String, refreshInterval: Int?,
-         strategies: [Strategy]) {
-        
-        self.registerService = registerService
-        self.toggleRepository = toggleRepository
-        self.appName = appName
-        self.url = url
-        self.refreshInterval = refreshInterval
-        self.strategies = strategies
-        
-        register(body: clientRegistration)
-            .then { _ in
-                self.toggleRepository.get(url: URL(string: url)!)
-            }
-            .catch { error in
-                log("error \(error)")
-        }
+  }
+  
+  // MARK: - Private Methods
```

Comment:
So this wasn't set up to have private and public methods set to a particular section. It was ordered so that methods are ordered in such a way so that they read like a story that doesn't cause the reader to need to jump around the code too much

Reply:
Ahhh okay, I can see that now that you mention it.